### DISCUSSION TITLE
airflow task going to queued -> failed with no logs

### DISCUSSION THREAD

**Discussion by rkarthik29** [apache/airflow#37077] [Q&A]
we have airflow version 2.6.2 and are using celery executor to run some long running tasks . The infrastructure is on ECS. We have multiple tasks run the workers. We have 3 tasks groups, with the celery workers in each group listentening on a different queue (default,small,large). The large(long running tasks) workers are set with a concurrency of 8, and the small (short running tasks) workers have a concurrency of 16. The default concurrency is 16 for the default queue. There is a large and small pool as well with 200 slots each. The operators set pool and queue to large or small based on the file size that needs to be processed using a branch operator. if the file is large, the brach operator triggers the large dag which pushes tasks to large queue and small dag if the files are small. All of this is working fine, but there are always tasks that get queued, but the moment they queued, they fail and there is not logs at all for failure. if i run 8 taks, which is the worker_concurrency , most of time 7 will run, 1 will go to queue and then fail. there are no logs in cloudwatch for that task. setting retries=True retries the tasks after 10 seconds and executes fine. We are trying to understand why the task is going to failed, instead of being in the queue. is it because there are more than one celery worker listen to a queue ( we have auto scaling on the workers ecs task def). Let me know what other info can help in debugging this issue.

**Comment by rkarthik29** [comment]
[[34m2024-01-29T23:26:20.609+0000[0m] {[34mscheduler_job_runner.py:[0m761} ERROR[0m - Executor reports task instance <TaskInstance:process_data.process_file manual__2024-01-29T18:25:39-05:00 map_index=14 [queued]> finished (failed) although the task says its queued. (Info: None) Was the task killed externally?[0m | [[34m2024-01-29T23:26:20.609+0000[0m] {[34mscheduler_job_runner.py:[0m761} ERROR[0m - Executor reports task instance <TaskInstance: process_data.process_file manual__2024-01-29T18:25:39-05:00 map_index=14 [queued]> finished (failed) although the task says its queued. (Info: None) Was the task killed externally?[0m
-- | --

i see these above errors in the scheduler.

**Reply by potiuk** [reply]
Likely your infrastructure is killing the tasks - but for what reason - you need to check in ECS. Memory, temporary failures. lack  of other resources are usually the culprit, but you need to check your deployment logs (all of them) to find correlated events and logs showing the reason why something is killing your tasks.

I'd say - most probable reason is that you do not have enough memory and the task tries to allocate it and gets OOM - this one gives no chances to even write in the logs what's happened because OOM is done by kernel and kills such task immediately. You need to thoroughly examine your deployment logs.

**Comment by rkarthik29** [comment]
We have 120GB memory  , so it is not memory for sure, per cloudwatch we are only using   .72% of available ram.

**Reply by potiuk** [reply]
Then you need to look deeper. That was a wild guess but only you can find out in your ECS if there are any corellated events. No-one else can do it.

**Comment by rkarthik29** [comment]
looks like the workers instances were being deployed in west region as well and were becoming part of the celery workers listening to the queue, but the dags which are in an efs were not accessible to these nodes and that was causing the error.
